### PR TITLE
add resonator Props

### PR DESCRIPTION
**Pull request by techmannih**
add resonator props
@imrishabh18 please check

### CODE REVIEW

**Review by imrishabh18 — CHANGES_REQUESTED** [review]
Another thing, missed it in the comment before. The ResonatorProps needs to be exported as well from the `index.ts` file.

cc: @seveibar Does these properties look good for resonator? We need it for the arduino nano - https://github.com/tscircuit/core/issues/426

File: lib/components/resonator.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  pinVariant: "3pin"
+}
+
+export const resonatorProps = commonComponentProps.extend({
+  frequency: frequency,
+  loadCapacitance: capacitance,
+  pinVariant: z.literal("3pin"),
+})
+
+export const resonatorPins = lrPins
```
Comment (seveibar) [inline lib/components/resonator.ts:22]: This should be removed or changed to reflect the 3 pins, lrPins refers to left/right pins which isnt appropriate for a 3 pin circuit.
```suggestion
```

**Review by seveibar — APPROVED** [review]

**Comment by seveibar** [comment]
Approved after rishabhs feedback is addressed!

**Review by imrishabh18 — APPROVED** [review]

**Comment by imrishabh18** [comment]
@techmannih The bounty is there for the whole issue https://github.com/tscircuit/core/issues/426 which includes this as one of the task.

**Pull request merged**